src/lib/models/ConfigModel.ts

```diff
@@ -1,172 +1,198 @@
-    #                           Default: 'Moderate'
-    #     'tags'              - [Optional] The string array of tag values to apply to this rule by default.
-    #                           Default: ['Recommended']
-    #
-    # [Example usage]:
-    # engines:
-    #   regex:
-    #     custom_rules:
-    #       NoTodoComments:
-    #         regex: /\\/\\/[ \\t]*TODO/gi
-    #         file_extensions: [".cls", ".trigger"]
-    #         description: "Prevents TODO comments from being in apex code."
-    #         violation_message: "A comment with a TODO statement was found. Please remove TODO statements from your apex code."
-    #         severity: "Info"
-    #         tags: ['TechDebt']
-    custom_rules: {} # Remove this empty object {} when you are ready to define your first custom rule
-
-  # Custom configuration settings for the 'retire-js' engine
-  # See <LINK_COMING_SOON> to learn more about these settings.
-  retire-js:
-
-    # Whether to turn off the 'retire-js' engine so that it is not included when running Code Analyzer commands
-    disable_engine: false`;
+function getDefaultRule(defaultContext: ConfigContext, engineName: string, ruleName: string): Rule|null {
+	try {
+		return defaultContext.rules.getRule(engineName, ruleName);
+	} catch (e) {
+		// istanbul ignore next
+		return null;
+	}
+}
+
+function toYamlRule(userRule: Rule, defaultRule: Rule|null, styled: boolean): string {
+	const ruleName: string = userRule.getName();
+	const userSeverity: SeverityLevel = userRule.getSeverityLevel();
+	const userTags: string[] = userRule.getTags();
+
+	let severityYaml = yaml.dump({severity: userSeverity});
+	let tagsYaml = yaml.dump({tags: userTags});
+
+	if (defaultRule != null) {
+		const defaultSeverity: SeverityLevel = defaultRule.getSeverityLevel();
+		const defaultTagsJson: string = JSON.stringify(defaultRule.getTags());
+
+		if (userSeverity !== defaultSeverity) {
+			const comment = getMessage(BundleName.ConfigModel, 'template.modified-from', [defaultSeverity]);
+			severityYaml = severityYaml.replace('\n', ` ${toYamlComment(comment, styled)}\n`);
+		}
+
+		if (JSON.stringify(userTags) !== defaultTagsJson) {
+			const comment = getMessage(BundleName.ConfigModel, 'template.modified-from', [defaultTagsJson]);
+			// The YAML spec requires a trailing newline, so we know that there's at least one newline somewhere in the
+			// string. If we inject a comment before the first newline we encounter, then it will look clean.
+			tagsYaml = tagsYaml.replace('\n', ` ${toYamlComment(comment, styled)}\n`);
+		}
+	}
+	return `"${ruleName}":\n${indent(severityYaml, 2)}${indent(tagsYaml, 2)}`;
+}
+
+function toYamlEngines(relevantEngines: Set<string>, userContext: ConfigContext, styled: boolean): string {
+	if (userContext.rules.getCount() === 0) {
+		const comment = getMessage(BundleName.ConfigModel, 'template.yaml.no-engines-selected');
+		return `engines: {} ${toYamlComment(comment, styled)}`;
+	}
```

Comment:
This will prevent the config from being shown when disable_engine: true... so I think you need to remove this condition.

Reply:
No, it'll only do that if every engine is disabled, or if the rule selection was such gibberish that no rules were returned at all. if no rules were returned at all, then we need special handling like this, because otherwise we'll end up with engines:\n and no subentries, which is invalid YAML.

Reply:
But if a user does `sf code-analyzer config -r eslint` when eslint's engine is disabled... they still want to see eslint config information.

Reply:
OK... to handle the no engine case... then switch the condition to instead be on line 170
```
	if (relevantEngines.size === 0) {
	```
	instead